src/main/java/jarvis/logic/commands/AddNoteCommand.java

```diff
@@ -80,12 +80,12 @@ private String executeAddOverallNote(Lesson lessonToAdd) {
         return String.format(MESSAGE_ADD_OVERALL_NOTE_SUCCESS, lessonToAdd, note);
     }
 
-    private String executeAddStudentNote(Model model, Lesson lessonToAdd) throws CommandException {
-        List<Student> lastShownStudentList = model.getFilteredStudentList();
-        if (studentIndex.getZeroBased() >= lastShownStudentList.size()) {
+    private String executeAddStudentNote(Lesson lessonToAdd) throws CommandException {
+        Student[] studentsInLesson = lessonToAdd.getStudents().toArray(new Student[0]);
+        if (studentIndex.getZeroBased() >= studentsInLesson.length) {
             throw new CommandException(Messages.MESSAGE_INVALID_STUDENT_DISPLAYED_INDEX);
         }
-        Student studentToAdd = lastShownStudentList.get(studentIndex.getZeroBased());
+        Student studentToAdd = studentsInLesson[studentIndex.getZeroBased()];
```

Comment:
Instead of doing this, `Lesson` should have a `getStudent` method that takes in the student index and returns the student. Same applies for other commands that use student indexes.